tools/mason/MasonNew.chpl

```diff
@@ -91,13 +112,153 @@ proc masonNew(args) throws {
   }
 }
 
+/*
+  Starts an interactive session to create a
+  new library project.
+*/
+proc beginInteractiveSession(defaultPackageName: string) throws {
+  var isInit: bool = false;
+  if defaultPackageName != '' then isInit = true;
+  if isInit && isFile('./Mason.toml') {
+    throw new owned MasonError("Unable to run mason init --interactive. Mason.toml" + 
+                               " file already exists.");
+    exit(1);
+  }
+  writeln("""This is an interactive session to walk you through creating a library
+project using Mason. The following queries covers the common items required to
+create the project. Suggestions for defaults are also provided which will be
+considered if no input is given.""");
+  writeln();
+  writeln("Press ^C to quit interactive mode.");
+  var packageName: string;
+  var defPackageName: string = defaultPackageName;
+  var version: string;
+  var defaultVersion: string = "0.1.0";
+  var chapelVersion: string;
+  var currChapelVersion = getChapelVersionStr();
+  var gotCorrectPackageName = false;
+  var gotCorrectPackageVersion = false;
+  var gotCorrectChapelVersion = false;
+  while(1){
+    try {
+      if !gotCorrectPackageName {
+        write("Package name ");
+        if isInit then write("(" + defPackageName + ") ");
+        write(": ");
+        IO.stdout.flush();
+        IO.stdin.readline(packageName);
+        if packageName == '' {
```

Comment:
Does it make sense to wrap this check in a function, e.g. `exitOnEOF(packageName);` or `checkEOF(packageName);`

Reply:
Okay, I think it'd be much cleaner.

Reply:
BTW - nice trick you found there. Works like a charm!